tests/test_manage_certs.py

```diff
@@ -0,0 +1,88 @@
+"""Unit tests for the manage_certs.py script."""
+
+# To run the tests:
+#   PYTHONPATH=../templates python3 -m unittest test_manage_certs.py
+# To lint:
+#   PYTHONPATH=../templates pylint3 --max-line-length=120 --disable=missing-docstring --reports=n test_manage_certs.py
+
+import argparse
+import collections
+import io
+import logging
+import unittest
+from unittest import mock
+
+import manage_certs
+
+
+ALL_DOMAINS = collections.OrderedDict([
+    ("some.domain.name", "be-corresponding-backend"),
+    ("weird-whitespace.domain.name", "be-corresponding-backend"),
+    ("uppercase.domain.name", "be-corresponding-backend"),
+    ("different.backend.domain", "be-differenct-backend"),
+    ("yet.another.backend.domain", "be-UPPERCASE-backend"),
+])
+
+
+class TestManageCerts(unittest.TestCase):
+
+    def setUp(self):
+        self.config = argparse.Namespace()
+
+    def test_get_all_domain(self):
+        self.config.haproxy_backend_map = mock.MagicMock()
+        self.config.haproxy_backend_map.open.return_value.__enter__.return_value = io.StringIO(
+            "# This is a comment.\n"
+            "\n"
+            "some.domain.name be-corresponding-backend\n"
+            "  weird-whitespace.domain.name \t be-corresponding-backend  \n"
+            "UPPERCASE.domain.name be-corresponding-backend\n"
```

Comment:
@smarnach This is a case where writing an isolated test for this would really help to point out the issue, and have clearer errors if that test ever fails (since the name of the test would describe the bug). Since you've already written the tests all at once now, I won't ask you to change it here - but a comment to highlight the issue and not lose the context/test in the future would still be good. Could you consider it when fixing bugs in the future though?